pkg/build/generator/generator.go

```diff
@@ -353,6 +363,13 @@ func (g *BuildGenerator) Clone(ctx kapi.Context, request *buildapi.BuildRequest)
 	newBuild := generateBuildFromBuild(build, buildConfig)
 	glog.V(4).Infof("Build %s/%s has been generated from Build %s/%s", newBuild.Namespace, newBuild.ObjectMeta.Name, build.Namespace, build.ObjectMeta.Name)
 
+	// Add the original Build's OwnerReferences to the new Builds. This will
+	// make sure that the new Build is deleted when the original BuildConfig
+	// is deleted.
+	if len(build.OwnerReferences) > 0 {
```

Comment:
Just to confirm, the logic here is: OpenShift will always add OwnerReferences to builds instantiated from buildconfigs; if I happen to remove the OwnerReference manually and then clone that build, the resulting build will also not have an OwnerReference?